### DISCUSSION TITLE
Team Member Permissions

### DISCUSSION THREAD

**Discussion by vishalkadam47** [coollabsio/coolify#3232] [Improvement Requests]
### Description

The current permission structure for team grants the same level of deletion authority to all team members, irrespective of their role (owner, admin, or regular member). This means that any team member has the authority to delete any deployed service or project, regardless of who created it.

### Minimal Reproduction (if possible, example repository)

Proposed Permission Structure:

Owner:
- Has full access to manage all services and projects.
- Can create, edit, delete, deploy, and view all projects.
- Deployed services and projects remain hidden by default until explicitly exposed by the owner.

Admin:
- Can manage and delete services and projects created by members, but not those created by the owner.

Member:
- Can only delete their own deployments.
- Can view and collaborate on assigned projects.

## Bonus: Consider a "User" role for a Non-Technical Member.

- Can deploy required Template
- Delete own deployments.
- limited side bar items

# Deployed services and projects remain hidden by default unless granted specific access by owner or admin.

### Version

v4.0.0-beta.285

**Comment by lntcT** [comment]
Id need this to work as expected as well.
The possibility of any team member having root level privileges (especially through "command center") is too much of a concern to move everything to coolify for me.

**Comment by pjcdz** [comment]
I'd love this to be solved soon ^^

**Comment by horacio-pedro** [comment]
In fact, this is a situation that has also bothered me.

It is a very dangerous hierarchical logic flaw, especially for companies that use Coolify as a solution, I hope that this wonderful team that provides us with such a useful tool like this can fix this bug.

**Comment by humongouselephant** [comment]


**Reply by wundo** [reply]
Please, this would be awesome!

**Comment by peaklabs-dev** [comment]
@vishalkadam47 This is a new feature not a bug so I am migrating this to a discussion.

**Comment by thewilloftheshadow** [comment]
This is a major deal breaker for us, without having a proper permissions system we can't use Coolify for our services

**Comment by Mase3206** [comment]
A regular user being able to delete a team they are not a manager or owner of is a bug. This may be considered expected behavior by the devs, but it is absolutely unexpected and preventable. What's the point of giving users the "Member", "Manager", or "Owner" labels if they don't do anything?

**Reply by vishalkadam47** [reply]
that's the reason I have created this as issue in detailed but Collaborator - @peaklabs-dev has moved this issue to discussion and its been around 4 months, I guess this issues has been ignored.
I did try to look into the code but its PHP and its hard for me to understand.

**Reply by peaklabs-dev** [reply]
It will come when it is ready. I agree that we have a lot to work on with the permissions, but it is such a huge change probably several thousand lines and we are now focusing on releasing v4 stable first before we make such huge changes.

**Comment by fedy97** [comment]
+1 on this

**Comment by wundo** [comment]
+1 on this

**Comment by sanjaybora04** [comment]
Instead of giving roles, allow admin to select permissions, this gives much more flexibility and easier to implement

**Comment by johndobiedesign** [comment]
+1 on this also!

**Comment by dentroai** [comment]
for now the only workaround is to create a separate "production" Team where I don't invite team "members" right?

**Comment by Shadow1363** [comment]
+1 on this!
